src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -647,8 +647,12 @@ private static unsafe nuint GetIndexOfFirstNonAsciiChar_Sse2(char* pBuffer, nuin
             uint SizeOfVector128InBytes = (uint)Unsafe.SizeOf<Vector128<byte>>();
             uint SizeOfVector128InChars = SizeOfVector128InBytes / sizeof(char);
 
-            Debug.Assert(Sse2.IsSupported, "Should've been checked by caller.");
-            Debug.Assert(BitConverter.IsLittleEndian, "SSE2 assumes little-endian.");
+            Debug.Assert(Sse2.IsSupported || AdvSimd.Arm64.IsSupported, "Sse2 or AdvSimd64 required.");
+            Debug.Assert(BitConverter.IsLittleEndian, "This SSE2/Arm64 implementation assumes little-endian.");
+
+            Vector128<byte> bitmask = BitConverter.IsLittleEndian ?
+                Vector128.Create(0x80402010_08040201).AsByte() :
+                Vector128.Create(0x01020408_10204080).AsByte();
```

Comment:
I think this is also true for other PRs that Carlos sent, but can you remind me why we use `bitMask` for `!IsLittleEndian` if it is not supported? Is the idea that when we start supporting, the code will just work?

Reply:
Yup, that's the idea.